File: product.go
package main

import (
	"time"

	"github.com/google/uuid"
)

type Product struct {
	ID          uuid.UUID
	Name        string
	Slug        string
	ImagePath   string
	Description string
	CategoryID  uuid.UUID
	Variants    []Variant
}

type Variant struct {
	ID            uuid.UUID
	Name          string
	ProductID     uuid.UUID
	Price         float64
	StockQuantity int32
	ImageUrl      string
	VariantName   string
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

File: frontend/src/types/product.ts
import { Breadcrumb } from './global';

export type Variant = {
  id: string;
  name: string;
  price: number;
  stockQuantity: number;
  imageUrl: string;
  variantName: string;
};

export type Product = {
  id: string;
  name: string;
  slug: string;
  imagePath: string;
  description: string;
  variants: Variant[];
};

export type ProductResponse = {
  product: Product;
  breadcrumbs: Breadcrumb[];
};
